osf/metrics/counted_usage.py

```diff
@@ -157,15 +157,25 @@ def _get_provider_id(guid_referent):
     if isinstance(provider, str):
         return provider         # quacks like BaseFileNode
     elif provider:
-        return provider._id     # quacks like Registration, Preprint
-    return 'osf'                # quacks like Node
+        return provider._id     # quacks like Registration, Preprint, Collection
+    return 'osf'                # quacks like Node, Comment, WikiPage
 
 
-def _get_surrounding_guids(guid_referent):
-    immediate_wrapper = (
-        getattr(guid_referent, 'target', None)          # quacks like BaseFileNode
-        or getattr(guid_referent, 'parent_node', None)  # quacks like AbstractNode
+def _get_immediate_wrapper(guid_referent):
+    if hasattr(guid_referent, 'verified_publishable'):
+        return None                                     # quacks like Preprint
+    return (
+        getattr(guid_referent, 'parent_node', None)     # quacks like AbstractNode
+        or getattr(guid_referent, 'node', None)         # quacks like WikiPage, Comment
+        or getattr(guid_referent, 'target', None)       # quacks like BaseFileNode
     )
-    if immediate_wrapper:
-        return [immediate_wrapper._id, *_get_surrounding_guids(immediate_wrapper)]
-    return []
+
+def _get_surrounding_guids(guid_referent):
+    surrounding_guids = []
+    current_referent = guid_referent
+    while current_referent:
+        next_referent = _get_immediate_wrapper(current_referent)
+        if next_referent:
```

Comment:
So, just to be clear, if we're starting with a Wiki/File/etc., we'll grab the node that it belongs to and then all of the nodes in its hierarchy from there?

Reply:
yep, exactly